src/coordinator/storage/local/config.go

```diff
@@ -0,0 +1,190 @@
+	var (
+		numUnaggregatedClusterNamespaces int
+		numAggregatedClusterNamespaces   int
+		unaggregatedClusterNamespaceCfg  = &unaggregatedClusterNamespaceConfiguration{}
+		aggregatedClusterNamespacesCfgs  []*aggregatedClusterNamespacesConfiguration
+		unaggregatedClusterNamespace     UnaggregatedClusterNamespaceDefinition
+		aggregatedClusterNamespaces      []AggregatedClusterNamespaceDefinition
+	)
+	for _, clusterCfg := range c {
+		client, err := clusterCfg.newClient(client.ConfigurationParameters{})
+		if err != nil {
+			return nil, err
+		}
+
+		aggregatedClusterNamespacesCfg := &aggregatedClusterNamespacesConfiguration{
+			client: client,
+		}
+
+		for _, n := range clusterCfg.Namespaces {
+			switch n.StorageMetricsType {
+			case storage.UnaggregatedMetricsType:
+				numUnaggregatedClusterNamespaces++
+				if numUnaggregatedClusterNamespaces > 1 {
+					return nil, fmt.Errorf("only one unaggregated cluster namespace  "+
+						"can be specified: specified %d", numUnaggregatedClusterNamespaces)
+				}
+
+				unaggregatedClusterNamespaceCfg.client = client
+				unaggregatedClusterNamespaceCfg.namespace = n
+
+			case storage.AggregatedMetricsType:
+				numAggregatedClusterNamespaces++
+
+				aggregatedClusterNamespacesCfg.namespaces =
+					append(aggregatedClusterNamespacesCfg.namespaces, n)
+
+			default:
+				return nil, fmt.Errorf("unknown storage metrics type: %v",
+					n.StorageMetricsType)
+			}
+		}
+
+		if len(aggregatedClusterNamespacesCfg.namespaces) > 0 {
+			aggregatedClusterNamespacesCfgs =
+				append(aggregatedClusterNamespacesCfgs, aggregatedClusterNamespacesCfg)
+		}
+	}
+
+	if numUnaggregatedClusterNamespaces != 1 {
+		return nil, fmt.Errorf("one unaggregated cluster namespace  "+
+			"must be specified: specified %d", numUnaggregatedClusterNamespaces)
+	}
+
+	// Connect to all clusters in parallel
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		cfg := unaggregatedClusterNamespaceCfg
+		cfg.result.session, cfg.result.err = cfg.client.DefaultSession()
+	}()
+	for _, cfg := range aggregatedClusterNamespacesCfgs {
+		cfg := cfg // Capture var
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			cfg.result.session, cfg.result.err = cfg.client.DefaultSession()
+		}()
+	}
+
+	// Wait
+	wg.Wait()
+
+	if unaggregatedClusterNamespaceCfg.result.err != nil {
+		return nil, fmt.Errorf("could not connect to unaggregated cluster: %v",
+			unaggregatedClusterNamespaceCfg.result.err)
+	}
+
+	unaggregatedClusterNamespace = UnaggregatedClusterNamespaceDefinition{
+		NamespaceID: ident.StringID(unaggregatedClusterNamespaceCfg.namespace.Namespace),
+		Session:     unaggregatedClusterNamespaceCfg.result.session,
+		Retention:   unaggregatedClusterNamespaceCfg.namespace.Retention,
+	}
+
+	for i, cfg := range aggregatedClusterNamespacesCfgs {
+		if cfg.result.err != nil {
+			return nil, fmt.Errorf("could not connect to aggregated cluster #%d: %v",
```

Comment:
Might be better to have the invalid cluster ID rather than the number? Deduping could mess up the order and make this message less useful

Reply:
There is no such thing as a cluster ID, but I can put the namespace (although if you share namespaces across different clusters you still can't differentiate it), so I'll add the namespace next to number.

Reply:
Ah that's unfortunate. Any way we can specify which failed to connect if we've deduped any?